### PR TITLE
complex stepping docs

### PR DESCRIPTION
**Pull request by Datseris**

### CODE REVIEW

**Review by Libbum — APPROVED** [review]

**Comment by Libbum** [comment]
Looks good. 

@fbanning, anything you think is missing?

**Comment by Libbum** [comment]
Actually, maybe one comment: 
This suggests a `scheduler1` and `scheduler2`, although our `model.scheduler` is always only one of these. Is there a need to consider this design choice at all? I know we need a *default* scheduler of course, but perhaps we also need to adjust some language in the scheduler docs too.

**Comment by Libbum** [comment]
Great!

File: docs/src/tutorial.md [hunk 1]
```diff
@@ -75,6 +75,31 @@ dummystep
     An example can be seen in the `model_step!` function of [Daisyworld](@ref),
     where a `tick` is increased at each step.
 
+### Advanced stepping
+The interface of [`step!`](@ref), which allows the option of both `agent_step!` and `model_step!` is driven mostly by convenience. In principle, the `model_step!` function by itself can perform all operations related with stepping the ABM.
+However, for many models, this simplified approach offers the benefit of not having to write an explicit loop over existing agents inside the `model_step!`.
+Most of the examples in our documentation can be expressed using an independent `agent_step!` and `model_step!` function.
+
+On the other hand, more advanced models require special handling for scheduling, or many need to schedule several times and act on different subsets of agents with different functions.
```
Comment (fbanning) [inline docs/src/tutorial.md:83]: > many

may

Comment (Libbum) [reply docs/src/tutorial.md:83]: Fixed in 2be5d13e

**Pull request merged**